smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -463,29 +463,33 @@ def pollChildren(child = null) {
 }
 
 // Poll Child is invoked from the Child Device itself as part of the Poll Capability
-def pollChild(child){
-
-	if (pollChildren(child)){
-		if (!child.device.deviceNetworkId.startsWith("ecobee_sensor")){
-			if(atomicState.thermostats[child.device.deviceNetworkId] != null) {
-				def tData = atomicState.thermostats[child.device.deviceNetworkId]
-				log.info "pollChild(child)>> data for ${child.device.deviceNetworkId} : ${tData.data}"
-				child.generateEvent(tData.data) //parse received message from parent
-			} else if(atomicState.thermostats[child.device.deviceNetworkId] == null) {
-				log.error "ERROR: Device connection removed? no data for ${child.device.deviceNetworkId}"
-				return null
+def pollChild(){
+
+	def devices = getChildDevices()
+
+	if (pollChildren()){
+		devices.each { child ->
+			log.info "***found $child"
+			if (!child.device.deviceNetworkId.startsWith("ecobee_sensor")){
```

Comment:
This is the same premise as above, no need to only poll for a single child device, instead the API will return information on all of the child devices. Which means that it is more efficient to update the information for all of the children at the same time as well.

The nice side effect of this is if you were to manually use "refresh" from the Things page that it will not only refresh the data for that sensor/thermostat but will also update the data for all of the other sensors/thermostats at the same time! This also means a lot fewer API calls are needed.